src/helpers/defaultRules.js

```diff
@@ -0,0 +1,255 @@
+'use strict'
+const defaultRuleQueries = require('@database/queries/defaultRule')
+const mentorQueries = require('@database/queries/mentorExtension')
+const menteeQueries = require('@database/queries/userExtension')
+const { Op } = require('sequelize')
+
+const { isAMentor } = require('@generics/utils')
+
+class MissingFieldError extends Error {
+	constructor(field) {
+		super(`Value for field ${field} is undefined or null`)
+		this.name = 'MissingFieldError'
+		this.field = field
+	}
+}
+
+/**
+ * Gets the valid configurations based on user roles.
+ *
+ * @param {Array<Object>} config - The configuration array.
+ * @param {Array<Object>} userRoles - The user roles array.
+ * @returns {Array<Object>} - The array of valid configurations.
+ */
+function getValidConfigs(config, userRoles) {
+	const userRoleTitles = userRoles.map((role) => role.title)
+	const validConfigs = []
+
+	for (let conf of config) {
+		const { requester_roles, role_config } = conf
+		const { exclude } = role_config
+
+		if (exclude) {
+			// Exclude logic: valid if none of the user roles are in the requester_roles
+			const hasExclusion = requester_roles.some((role) => userRoleTitles.includes(role))
+			if (!hasExclusion) {
+				validConfigs.push(conf)
+			}
+		} else {
+			// Include logic: valid if any of the user roles are in the requester_roles
+			const hasInclusion = requester_roles.some((role) => userRoleTitles.includes(role))
+			if (hasInclusion) {
```

Comment:
@nevil-mathew  
this line is repeated
 const hasInclusion = requester_roles.some((role) => userRoleTitles.includes(role))
change the logic to avoid that